lib/core/services/swap_service.dart

```diff
@@ -0,0 +1,80 @@
+import 'dart:convert';
+
+import 'package:flutter_dotenv/flutter_dotenv.dart';
+
+import '../app/logger.dart';
+import 'package:http/http.dart' as http;
+
+class SwapService {
+  final Map<String, String> header = {
+    "Content-type": "application/json",
+    "Accept": "application/json"
+  };
+  static final baseUrl = dotenv.env['WEB3_API_URL'] ?? "";
+
+  // "http://10.0.2.2:8080/api/v1/web3";
+
+  Future<Map<String, dynamic>> getExchangeRate(String private,
+      String tokenAddress1, String tokenAddress2, double amountToken1) async {
+    Uri url = Uri.parse("$baseUrl/pairrate");
+    Map<String, dynamic> body = {
+      "private": private,
+      "token_address0": tokenAddress1,
+      "token_address1": tokenAddress2,
+      "amount_token0": amountToken1.toStringAsFixed(0),
+    };
+    var encodedBody = jsonEncode(body);
+    var result = await http.post(url, body: encodedBody, headers: header);
+
+    var res = jsonDecode(result.body);
+    print(res);
+    return {
+      "token1": res['pair'][0],
+      "token2": res['pair'][1],
+    };
+  }
+
+  Future<void> swapExactEthForToken(
```

Comment:
Use named param

Reply:
i dont get what you mean by that ? whats the problem